src/networking/http-client.ts

```diff
@@ -15,15 +15,27 @@ export async function performRequest<RequestBody, ResponseType>(
   body?: RequestBody,
   headers?: { [key: string]: string },
 ): Promise<ResponseType> {
-  const response = await fetch(endpoint.url(), {
-    method: endpoint.method,
-    headers: getHeaders(apiKey, headers),
-    body: getBody(body),
-  });
+  try {
+    const response = await fetch(endpoint.url(), {
+      method: endpoint.method,
+      headers: getHeaders(apiKey, headers),
+      body: getBody(body),
+    });
 
-  await handleErrors(response, endpoint);
+    await handleErrors(response, endpoint);
 
-  return (await response.json()) as ResponseType; // TODO: Validate response is correct.
+    return (await response.json()) as ResponseType; // TODO: Validate response is correct.
+  } catch (error) {
+    if (error instanceof TypeError) {
```

Comment:
According to docs: https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#checking_that_the_fetch_was_successful. TypeError will be a network error.